### PR TITLE
Update rating API to sort by created time

### PR DESCRIPTION
**Pull request by nguyen356**
anh Check lai giup em voi nhe

### CODE REVIEW

File: src/modules/rating/controllers/admin-rating.controller.ts [hunk 1]
```diff
@@ -68,25 +68,29 @@ export class AdminRatingController {
...
+					sortOrder,
 				)
 			}
-
 			let newCursor: string | null = null
-
 			if (ratings.length > 0) {
 				if (ratings.length > size) {
-					newCursor = ratings[size]._id
+					newCursor = ratings[size].createdAt.toISOString() // Convert createdAt Date to ISO string
```
Comment (minhnguyenduy99) [inline src/modules/rating/controllers/admin-rating.controller.ts]: What is the purpose of this update ?

File: src/modules/rating/database/rating.repository.ts [hunk 2]
```diff
@@ -42,18 +44,18 @@ export class RatingRepository {
...
+		sortOrder: 'asc' | 'desc' = 'asc',
 	): Promise<Rating[]> {
 		try {
 			const query: any = { product_id, status }
 
 			if (cursor) {
 				query['_id'] = { $gte: cursor }
 			}
-
+			const sortCriteria = { _id: sortOrder === 'desc' ? -1 : 1 } // Sort by _id in descending order if sortOrder is 'desc', otherwise in ascending order
```
Comment (minhnguyenduy99) [inline src/modules/rating/database/rating.repository.ts]: Remove unused variable

File: src/modules/rating/controllers/admin-rating.controller.ts (hunk 1 above)
Comment (nguyen356) [reply src/modules/rating/controllers/admin-rating.controller.ts]: this update is chosing the lastest created date from that page and when change other page will continute to show the newest create date of the rating

File: src/modules/rating/database/rating.repository.ts (hunk 2 above)
Comment (nguyen356) [reply src/modules/rating/database/rating.repository.ts]: yes boss :b

File: src/modules/rating/controllers/admin-rating.controller.ts (hunk 1 above)
Comment (minhnguyenduy99) [reply src/modules/rating/controllers/admin-rating.controller.ts]: That's not the purpose of **newCursor**. **newCursor** is used to store the starting point of the query. For example:
- When user opens the rating page, it loads first 10 ratings. The starting point is _null_.
- When user clicks on **Next** button, the page should load next 10 ratings. The starting point should be the ID of 11th rating.

File: src/modules/rating/controllers/admin-rating.controller.ts [hunk 3]
```diff
@@ -68,26 +68,32 @@ export class AdminRatingController {
...
 				)
 			}
-
 			let newCursor: string | null = null
-
 			if (ratings.length > 0) {
 				if (ratings.length > size) {
-					newCursor = ratings[size]._id
-					ratings.splice(size)
+					newCursor = ratings[size - 1]._id.toString() // Use the ID of the last rating in the current page
```
Comment (minhnguyenduy99) [inline src/modules/rating/controllers/admin-rating.controller.ts]: Logic in this line is updated. Is the old version a bug ?

Comment (nguyen356) [reply src/modules/rating/controllers/admin-rating.controller.ts]: the old version not really bug just not suit the requirement so i change to set it stay choosing last rating of first page and when customer click next page (new cusor) the id been choosing will keep at top and still load the data from next page

Comment (minhnguyenduy99) [reply src/modules/rating/controllers/admin-rating.controller.ts]: If you choose last rating of the current page (let's say ID is **10**), the first rating of next page will be **10** as well, which is not the desired behavior.

**Pull request merged**